00_Base/src/services/credentials.service.ts

```diff
@@ -498,21 +499,32 @@ export class CredentialsService {
 
     const clientInformation = clientCpoTenant.clientInformation[0];
 
-    // await clientInformation.save(); // todo
-    await clientCpoTenant.save();
+    try {
+      await clientCpoTenant.save();
```

Comment:
In the future, maybe we can have a improvement for this endpoint that not always create a new cpo tenant but only update if it existing. And also not store this temp credentials A but only formal credentials C.

Reply:
Agreed, once we have multi tenant support, this would be adjusted to not create a new tenant but instead update if one already exists... I think we do need to store the temp credentials though because MSP then uses that token to authenticate with the CPO when making the requests

Reply:
Please correct me if my understanding is wrong. I don't think CPO need to store token A because this temp token A is generated by MSP. CPO use this token A to authenticate with MSP and get its version and version details and then send CPO's token B to MSP. Then MSP use token B to authenticate with CPO and then send CPO its formal token C.  I think even though we call this endpoint as register token A, it finally register and store the formal token C.

Reply:
Ah yes, regarding Token A, it does NOT need to be stored, we currently store it along with Token B since we need to create and save the record that has Token B as this is then used by MSP, I suppose that can be left out since it is temporary, and quickly overwritten after the response from MSP. Arguably, if something goes wrong, it may be good to see that value in the DB though and since the query is already being made to save the data, I am not sure if it hurts to leave that line there, what do you think?

Reply:
I had the idea to not store token A related things because storing and then remove them very soon are a bit of redundant. But I fully agree your concerns. Let's keep it as it is now.